onchain/rollups/contracts/dapp/CartesiDApp.sol

```diff
@@ -133,16 +134,14 @@ contract CartesiDApp is
             revert VoucherReexecutionNotAllowed();
         }
 
-        // execute voucher
-        (bool succ, ) = _destination.call(_payload);
+        // execute voucher, handles success and failure internally.
+        // If `_destination` reverts with a revert reason or custom
+        // error, it is bubbled up by this function.
+        Address.functionCall(_destination, _payload);
 
-        // if properly executed, mark it as executed and emit event
-        if (succ) {
-            voucherBitmask.setBit(voucherPosition, true);
-            emit VoucherExecuted(voucherPosition);
-        }
-
-        return succ;
+        // Succesfull execution is ensured previously, mark it as executed and emit event
```

Comment:
We already said in the previous line that failure is handled internally, so clearly program flow will only reach this line if the voucher was successfully executed.

```suggestion
        // mark it as executed and emit event
```